google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,260 @@
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new GetAccountHierarchy().runExample(googleAdsClient, params.managerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the examples.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param managerId managerCustomerId the customer ID.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, Long managerId) {
+    List<Long> seedCustomerIds = new ArrayList<>();
+    if (!(managerId == null)) {
+      seedCustomerIds.add(managerId);
+    } else {
+      // Issues a request for listing all accessible customers by this authenticated Google account.
+      try (CustomerServiceClient customerServiceClient =
+          googleAdsClient.getLatestVersion().createCustomerServiceClient()) {
+        ListAccessibleCustomersResponse accessibleCustomers =
+            customerServiceClient.listAccessibleCustomers(
+                ListAccessibleCustomersRequest.newBuilder().build());
+        System.out.println(
+            "No manager customer ID is specified. The example will print the "
+                + "hierarchies of all accessible customer IDs:");
+
+        for (String customerResourceName : accessibleCustomers.getResourceNamesList()) {
+          long customer =
+              customerServiceClient.getCustomer(customerResourceName).getId().getValue();
+          System.out.println(customer);
+          seedCustomerIds.add(customer);
+        }
+      }
+    }
+
+    List<Long> unprocessedCustomerIds = new ArrayList<>();
+    for (long seedCustomerId : seedCustomerIds) {
+      // Instantiates several variables used later.
+      Map<Long, List<CustomerClient>> customerIdsToChildAccounts = new HashMap<>();
+      CustomerClient rootCustomerClient = null;
+      int depth = 0;
+      // Adds the seed customer ID to the list of IDs to be processed.
+      unprocessedCustomerIds.add(seedCustomerId);
+
+      // Creates the Google Ads Service client.
+      try (GoogleAdsServiceClient googleAdsServiceClient =
```

Comment:
Do we need to change the `loginCustomerId` here? Say for instance `listAccessibleCustomers()` returns 3 different manager accounts, we'll need to set `loginCustomerId` differently for each call to `select client_customer`.

Reply:
I believe making the managerId param required makes this a moot point.